### PR TITLE
DRIVERS-2698: clarify that the unified test format is only intended for enterprise servers

### PR DESCRIPTION
**Pull request by baileympearson**
Please complete the following before merging:

- [ ] Update changelog.
- [x] Make sure there are generated JSON files from the YAML test files.
- [x] Test changes in at least one language driver.
- [x] Test these changes against all server versions and topologies (including standalone, replica set, sharded clusters, and serverless).

### CODE REVIEW

File: source/unified-test-format/unified-test-format.rst [hunk 1]
```diff
@@ -78,6 +78,10 @@ Iterable
   spec uses this as the return value for ``find`` and permit API flexibility
   rather than stipulate that a cursor object be returned directly.
 
+Server Compatibility
+--------------------
+
+The unified test format is intended to only run against enterprise servers.
```
Comment (jyemin) [inline source/unified-test-format/unified-test-format.rst:84]: Consider adding: `with enableTestCommand=true` (or whatever the syntax is)

Comment (baileympearson) [reply source/unified-test-format/unified-test-format.rst:84]: Is the suggestion to clarify which tests / syntax doesn't work against community servers?

Comment (jyemin) [reply source/unified-test-format/unified-test-format.rst:84]: Yes, it seems similar to the use case presented in DRIVERS-2698: if you try to run the tests without this they will fail (and in great numbers!).

**Review by jyemin — APPROVED** [review]

File: source/unified-test-format/unified-test-format.rst [hunk 2]
```diff
@@ -78,6 +78,11 @@ Iterable
   spec uses this as the return value for ``find`` and permit API flexibility
   rather than stipulate that a cursor object be returned directly.
 
+Server Compatibility
+--------------------
+
+The unified test format is intended to only run against enterprise servers.  Additionally, the 
+unified test format makes use of the `FailCommand <enableTestCommands>`_ feature of mongodb.  The server MUST be launched with the ``enableTestCommands`` parameter enabled.
```
Comment (jmikola) [inline source/unified-test-format/unified-test-format.rst:85]:
```suggestion
The unified test format is intended to only run against enterprise servers. Additionally, many tests make use of `Server Fail Points`_, so servers MUST be launched with the ``enableTestCommands`` parameter enabled.
```

**Review by jmikola — CHANGES_REQUESTED** [review]
See above.

**Review by jmikola — APPROVED** [review]

**Review by alcaeus — APPROVED** [review]

**Pull request merged**